src/anomalib/utils/sweep/config.py

```diff
@@ -169,3 +169,33 @@ def set_in_nested_config(config: DictConfig, keymap: list, value: Any) -> None:
         Replaced model config {'parent1': {'child1': 'a', 'child2': 1}, 'parent3': False, 'parent2': 'model1'}
     """
     get_from_nested_config(config, keymap[:-1])[keymap[-1]] = value
+
+
+def flattened_config_to_nested(config: dict[str, Any]) -> DictConfig:
+    """Convert a flattened config dictionary to a nested DictConfig object.
+
+    Args:
+        config: dict[str, Any]: Flattened config dictionary.
+
+    Examples:
+        >>> flattened_config = {
+                "parent1.child1": 'a',
+                "parent1.child2": 1,
+                "parent2": 'model1',
+            }
```

Comment:
My bad, it should be like the following:
```suggestion
    >>> flattened_config = {
    ...     "parent1.child1": 'a',
    ...     "parent1.child2": 1,
    ...     "parent2": 'model1',
    ... }
```